### ISSUE TITLE
Timeout issue

### ISSUE THREAD

**Issue by reuben-hestabit** [alexchantastic/figma-export#1]
I am getting a timeout error even after increasing the time. How can I fix this?

await page.waitForURL(
    "https://www.figma.com/files/recents-and-sharing/recently-viewed**",
  );
  
<img width="980" alt="Screenshot 2024-06-04 at 4 19 36 PM" src="https://github.com/alexchantastic/figma-export/assets/129954426/b4b485e5-4245-47c8-a85a-17dc8c871be4">

**Comment by alexchantastic** [comment]
Happy to help! The timeout seems to be occurring during the auth routine. This typically means that the script is unable to log into Figma. 

Are you trying to log in with an email and password or by using an auth cookie?

**Comment by reuben-hestabit** [comment]
Thank you for the prompt response Alex!

I am getting the same timeout error with both email/password and auth cookie

**Comment by alexchantastic** [comment]
Got it! Thanks for the info. There's a few things I'd like to confirm to help debug this.

First, making sure that your `.env` file is correctly formatted (let's pretend we're using email/password). It should look like this:

```sh
FIGMA_EMAIL="YOUR_EMAIL"
FIGMA_PASSWORD="YOUR_PASSWORD"
FIGMA_ACCESS_TOKEN="YOUR_ACCESS_TOKEN"
DOWNLOAD_PATH="/Users/anonymous/Downloads/"
WAIT_TIMEOUT=10000
```

The quotes are technically optional, but can help you discover if there are any characters that need to be escaped.

---

Second, we can confirm that the auth setup script is running properly. You can check this by running `npm run report` which will pop up the HTML report. If you click **authenticate**, it should give you a breakdown of the script's steps:

![image](https://github.com/alexchantastic/figma-export/assets/604167/e12833ec-2faa-4d4d-8266-7be5e8146562)

![image](https://github.com/alexchantastic/figma-export/assets/604167/fede8312-ecc2-4f35-9116-cce70af54052)

---

And finally, the auth script confirms that you're logged in by checking the URL that you arrive at after logging in. In all my tests, this has been `https://www.figma.com/files/recents-and-sharing/recently-viewed`.

Could you check by logging into Figma in your browser (maybe in private browsing) and seeing what URL it redirects you to when you successfully log in?

**Comment by reuben-hestabit** [comment]
I checked with console logs that the parameters from the .env files are getting passed correctly. The URL my browser redirects to after a successful login is - https://www.figma.com/files/team/{team ID}/recents-and-sharing/recently-viewed?fuid=1120688133257404787

**Comment by reuben-hestabit** [comment]
This is the report - 
<img width="1020" alt="Screenshot 2024-06-04 at 10 55 43 PM" src="https://github.com/alexchantastic/figma-export/assets/129954426/bd0c6273-70e4-4f77-bfea-bff38f440231">

**Comment by alexchantastic** [comment]
Ah that's it! I just pushed up a new version (https://github.com/alexchantastic/figma-export/releases/tag/v1.1.1) that checks for an element rather than the page URL to see if you're logged in correctly.

I've only been able to test on starter and professional plans so I'm guessing you're on a different tier which uses a different URL scheme after you land when logging in.

Let me know if the new version works!

**Comment by reuben-hestabit** [comment]
Awesome dude! It works now 👍 
I am on a professional plan as well.

I added a slight update to the code to make it work for me - 

`await expect(page.getByTestId("ProfileButton")).toBeVisible({ timeout: 50000 });`

Adding an increased timeout did the trick for me, I was doing some testing and figma loads the profile image as the last element even after login, specially when it comes across unexpected browser dimensions.

**Comment by alexchantastic** [comment]
Oh interesting. It does look like Figma has some responsive layout going on which shifts the profile button a bit. I'll look into an even more stable way to detect successful log in when I find a bit more time, but I'm glad it's working for you now. 👍

**Comment by reuben-hestabit** [comment]
Thank you for your help! This has been a great learning experience for me, as I tried playwright for the first time.
Let me know if you ever need any help/assistance in figma, I will be glad to be of help. I owe you one 😁

**Comment by alexchantastic** [comment]
Found some extra time today 😉 to roll in some of the changes you mentioned.

I'm now checking for the profile button to mount rather than be visible. This should hopefully solve some of the layout issues. Playwright does default to `1280x720` so I'm not sure why it doesn't show up all the time. 🤔 I also bumped the assertion timeout to `10000ms` just to give some additional buffer room.

If you have a chance, check out [v1.1.2](https://github.com/alexchantastic/figma-export/releases/tag/v1.1.2) to see if it consistently fixes the issue for you.
